### ISSUE TITLE
Prefer 'if not method_call( )' to 'if method_call( ) = abap_false'

### ISSUE THREAD

**Issue by PeregrinTooc** [SAP/styleguides#236]
In this [issue in codepal for abap](https://github.com/SAP/code-pal-for-abap/issues/362#issuecomment-919760682) the discussion is which of the following is more in the spirit of clean abap:

```ABAP
if not condition_is_fulfilled( ).
...
endif.
```
or 

```ABAP
if condition_is_fulfilled( ) = abap_false.
...
endif.
```

**Comment by lucasborin** [comment]
Same question for the positive conditions:

```abap
if condition_is_fulfilled( ).
...
endif.
```

or 

```abap
if condition_is_fulfilled( ) = abap_true.
...
endif.
```

?

**Comment by jdgx** [comment]
Hi all,

As mentioned in the codepal discussion my vote goes in both cases to the 'operatorless' version
```
if [not] condition_is_fulfilled( ).
...
endif.
```
not just because it's shorter, but mainly because it is closer to natural language and hence easier to read and easier to understand.

Best,
Jonathan

**Comment by ConjuringCoffee** [comment]
I agree with @jdgx.

**Comment by pokrakam** [comment]
I agree with the shorter syntax, but I think there should be a caveat that omitting the comparison is just a shorthand form for `IS NOT INITIAL` and care must be taken to ensure it really does return an ABAP Boolean space/`X`. 

Otherwise we could run into trouble: 
```abap
METHOD is_true. 
  result = 'N'. 
ENDMETHOD. 
...
IF is_true( ).   "==> True!
```
